app/src/test/java/com/jalasoft/routesapp/util/algorithm/RouteCalculatorTest.kt

```diff
@@ -0,0 +1,105 @@
+        val line3 = LinePath("LA", "Metro", points3, start3, end3, stops3)
+        hiltRule.inject()
+        lines = listOf(line1, line2, line3)
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near the given points then it returns a list of single lines`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52094, -68.12419)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52355, -68.12269)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        assertTrue(result.isNotEmpty())
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the line stops are near two lines then it returns a list of lines with two transport`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52094, -68.12419)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52442, -68.12036)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        if (result.isNotEmpty()) {
+            assertNotNull(result.first().connectionPoint)
+        } else {
+            fail()
+        }
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when the points are far away from any line`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52221, -68.12749)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52442, -68.12036)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+        assertTrue(result.isEmpty())
+    }
+
+    @Test
+    fun `Given a list of lines and an origin an destination point when there are one and two transportation lines then it returns a list with single and double lines`() {
+        val originPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52078, -68.12344)
+        val destinationPoint = RouteAlgorithmFakeData.coordinatesToLocation(-16.52455, -68.12269)
+        val minDistance = 200.0
+        val minDistanceBtwStops = 200.0
+
+        val result = RouteCalculator.calculate(lines, destinationPoint, originPoint, minDistance, minDistanceBtwStops)
+
+        val singleLineExist = result.filter {
+            it.connectionPoint == null
+        }
+        val doubleLineExists = result.filter {
+            it.connectionPoint != null
+        }
+        assertTrue(singleLineExist.isNotEmpty() && doubleLineExists.isNotEmpty())
```

Comment:
Not only assert not null, also check the value

Reply:
Added asserts for expected results